hail/python/hail/utils/frozendict.py

```diff
@@ -0,0 +1,37 @@
+from typing import TypeVar, Dict
+from collections.abc import Mapping
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+class frozendict(Mapping):
```

Comment:
I think Mapping can take type arguments `Mapping[T, U]`.

Reply:
`typing.Dict` supports this, but `typing.Mapping` doesn't seem to, unless I'm misreading. I'm new to the python type annotations stuff.